File: Birder/ClientApp/src/app/_models/UserProfileViewModel.ts
export interface UserProfileViewModel {
    userName: string;
    avatar: string;
    registrationDate: Date | string;
    isOwnProfile: boolean;
    followersCount: number;
    followingCount: number;
    isFollowing: boolean;
    // followers: NetworkUserViewModel[];
    // following: NetworkUserViewModel[];
}

export interface NetworkUserViewModel {
    userName: string;
    avatar: string;
    isFollowing: boolean;
    isOwnProfile: boolean;
}

File: Birder/ViewModels/UserProfileViewModel.cs
namespace Birder.ViewModels;

public class UserProfileViewModel
{
    public NetworkUserViewModel User { get; set; }
    public DateTime RegistrationDate { get; set; }
    public ObservationAnalysisViewModel ObservationCount { get; set; }
    public int FollowersCount { get; set; }
    public int FollowingCount { get; set; }
}